### PR TITLE
Detector-Competition-New: add anthropic api key detector

### PR DESCRIPTION
**Pull request by lc**
### Description:
This pull-request implements detection for Anthropic's Claude API Keys: https://docs.anthropic.com/claude/docs.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### Verification:
image

### CODE REVIEW

**Comment by lc** [comment]
How do I add a label? Not sure if I'm being stupid but I'm trying to add the `Hacktoberfest-Detector-Competition-New` label.

File: pkg/detectors/anthropic/anthropic.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+			res, err := client.Do(req)
+			if err == nil {
+				defer res.Body.Close()
+				if res.StatusCode == http.StatusBadRequest {
+					var resp response
+					if err = json.NewDecoder(res.Body).Decode(&resp); err != nil {
+						s1.VerificationError = fmt.Errorf("unexpected HTTP response body: %w", err)
+						continue
+					}
+					fmt.Println(resp.Error.Message)
```
Comment (zricethezav) [inline pkg/detectors/anthropic/anthropic.go:88]: @lc suggestion: remove this print statement
```suggestion
```

Comment (lc) [reply pkg/detectors/anthropic/anthropic.go:88]: Whoops, done.

**Comment by zricethezav** [comment]
@lc resolve conflicts, then this is good to go

**Review by zricethezav — APPROVED** [review]

**Pull request merged**